kiwi/system/setup.py

```diff
@@ -892,14 +908,18 @@ def _import_custom_scripts(self):
                 ]
             )
 
-    def _call_script(self, name):
+    def _call_script(self, name, option_list=None):
         script_path = os.path.join(self.root_dir, 'image', name)
         if os.path.exists(script_path):
+            options = option_list or []
             command = ['chroot', self.root_dir]
             if not Path.access(script_path, os.X_OK):
                 command += ['bash']
-            command += ['/image/' + name]
-            config_script = Command.call(command)
+            command += ['/image/' + name] + options
+            profile = Profile(self.xml_state)
+            caller_environment = os.environ
```

Comment:
**Bug!** (and your tests didn't caught it!). If you want to copy the environment, then you should not make just another reference to the same memory address, but actually copy the whole thing like so:

```python
import copy
...
caller_environment = copy.deepcopy(os.environ)
```

Right now you do not even need that parameter, as you modifying it globally:

```
>>> os.environ["BLAH"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'BLAH'
>>> call_env = os.environ
>>> call_env["BLAH"] = "blah"
>>> os.environ["BLAH"]
'blah'
```

Reply:
Thanks, yes makes a lot sense